core/errors/test-timeout-error.ts

```diff
@@ -0,0 +1,7 @@
+import { MatchError } from "../_errors";
+
+export class TestTimeoutError extends MatchError {
+  public constructor(testTimeout: number) {
+    super(null, null, `The test exceeded the given timeout of ${testTimeout}ms.`);
```

Comment:
Not sure about this @jamesrichford. I think it might be clearer to simply check `error instanceof MatchError || error instanceof TestTimeoutError`.. took me a few moments to work out why `null` and `null` were being passed up.